### PR TITLE
Add gradle-commons to simplify the project setup

### PR DESCRIPTION
**Pull request by Azurelol**
## Description

Add gradle-comons, refactoring the project setup to simplify.

## Changes
* ![ADD] `gradle-commons`
* ![UPDATE] `AppCenterPlugin setup

[NEW]:https://atlas-resources.wooga.com/icons/icon_new.svg "New"
[ADD]:http://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:http://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:http://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:http://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:http://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:http://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:http://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:http://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:http://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:http://resources.atlas.wooga.com/icons/icon_webGL.svg "Web:GL"

### CODE REVIEW

File: src/main/groovy/wooga/gradle/appcenter/AppCenterConventions.groovy [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package wooga.gradle.appcenter
+
+import com.wooga.gradle.PropertyLookup
+
+class AppCenterConventions {
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/appcenter/AppCenterConventions.groovy]: technically you can't simply rename the class as the original class `AppCenterConsts` is part of the public API and would mean a breaking API change.

File: src/main/groovy/wooga/gradle/appcenter/AppCenterPlugin.groovy [hunk 2]
```diff
@@ -66,55 +65,34 @@ class AppCenterPlugin implements Plugin<Project> {
...
+        // TODO: Implement in a way that it handles the default value and the parsing from env/property
+//        extension.defaultDestinations.convention(AppCenterConventions.defaultDestinations.getValueProvider(project,
+//            {
+//                def result = (List<Map<String, String>> )it
+//                if (result == null){
+//                    result =it.split(',').collect { ["name": it.trim()] }
+//                }
+//                result
+//            }))
+
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/appcenter/AppCenterPlugin.groovy]: I don't get what the issue is. The default value then must be a `String` and goes through the same mapping process as an env or property value.

File: src/main/groovy/wooga/gradle/appcenter/AppCenterPlugin.groovy [hunk 3]
```diff
@@ -66,55 +65,34 @@ class AppCenterPlugin implements Plugin<Project> {
...
+            AppCenterConventions.defaultDestinations
         }))
 
-        extension.retryCount.set(project.provider({
-            String rawRetryCount = (project.properties[AppCenterConsts.RETRY_COUNT_OPTION]
-                    ?: System.getenv()[AppCenterConsts.RETRY_COUNT_ENV_VAR]) as String
-
-            (rawRetryCount) ? Integer.parseInt(rawRetryCount) : AppCenterConsts.defaultRetryCount
-        }))
+        // TODO: Should these still call set? Or we can invoke convention here?
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/appcenter/AppCenterPlugin.groovy]: convention is good here. There was no convention in gradle 4.10 and I kept it backwards compatible.

File: src/main/groovy/wooga/gradle/appcenter/internal/DefaultAppCenterPluginExtension.groovy [hunk 4]
```diff
@@ -17,92 +17,10 @@
...
 class DefaultAppCenterPluginExtension implements AppCenterPluginExtension {
-    final Property<String> apiToken
-    final Property<String> owner
-    final Property<String> applicationIdentifier
-    final ListProperty<Map<String, String>> defaultDestinations
-    final Property<Boolean> publishEnabled
-    final Property<Long> retryTimeout
-    final Property<Integer> retryCount
 
     DefaultAppCenterPluginExtension(Project project) {
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/appcenter/internal/DefaultAppCenterPluginExtension.groovy]: you can update the constructor to no longer need the project reference. This is part of the internal API can be changed.

File: src/main/groovy/wooga/gradle/appcenter/tasks/AppCenterTaskSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2018-2021 Wooga GmbH
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/appcenter/tasks/AppCenterTaskSpec.groovy]: This class was generated today so the year must be 2022

**Review by Larusso — CHANGES_REQUESTED** [review]
Looks Ok all in all. I would double check the copyrights.
Some files have no at all.
So the rule should be for those witout one. Check in the git histrory the first and the last commit to that file as the copyright years. This can also be done later but if you have the time that would be cool

File: src/main/groovy/wooga/gradle/appcenter/AppCenterConventions.groovy (hunk 1 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/appcenter/AppCenterConventions.groovy]: I see.

File: src/main/groovy/wooga/gradle/appcenter/internal/DefaultAppCenterPluginExtension.groovy (hunk 4 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/appcenter/internal/DefaultAppCenterPluginExtension.groovy]: Understood.

File: src/main/groovy/wooga/gradle/appcenter/tasks/AppCenterTaskSpec.groovy (hunk 5 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/appcenter/tasks/AppCenterTaskSpec.groovy]: Got it.

File: src/main/groovy/wooga/gradle/appcenter/AppCenterPlugin.groovy (hunk 2 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/appcenter/AppCenterPlugin.groovy]: Fixed during call.

**Review by Larusso — APPROVED** [review]
No changes to the integration tests!!! Awesome.

**Comment by Azurelol** [comment]
> No changes to the integration tests!!! Awesome.

I am glad the property tests were thorough.

**Pull request merged**